src/main/java/com/rultor/agents/github/CommentsTag.java

```diff
@@ -94,44 +106,191 @@ public CommentsTag(final Github ghub) {
+     * @throws IOException If there is a problem.
+     */
+    private void logTagTooOld(final Issue issue, final String tag,
+        final List<DefaultArtifactVersion> previous) throws IOException {
+        issue.comments().post(
+            String.format(
+                // @checkstyle LineLength (1 line)
+                CommentsTag.PHRASES.getString("CommentsTag.version-too-low"),
+                tag, previous.toString()
+            )
+        );
+        Logger.info(
+            this, "tag %s must be greater than previous version %s",
+            tag, previous
+        );
+    }
+
+    /**
+     * Logs a comment because the tag is invalid.
+     * @param issue The Issue to add the comment to.
+     * @param tag The tag which is invalid.
+     * @throws IOException If there is a problem.
+     */
+    private void logInvalidTag(final Issue issue, final String tag)
+        throws IOException {
+        issue.comments().post(
+            String.format(
+                CommentsTag.PHRASES.getString("CommentsTag.invalid-tag"), tag
+            )
+        );
+        Logger.info(this, "tag %s is invalid", tag);
+    }
+
+    /**
+     * Logs a comment because the tag has already been released.
+     * @param issue The Issue to add the comment to.
+     * @param tag The tag that has already been released.
+     * @param rels The other releases.
+     * @param home The URI of home.
+     * @throws IOException If there is a problem.
+     * @checkstyle ParameterNumberCheck (15 lines)
+     */
+    private void logDuplicateTag(final Issue.Smart issue, final String tag,
+        final Releases.Smart rels, final URI home) throws IOException {
+        final Release.Smart rel = new Release.Smart(rels.find(tag));
+        rel.body(
+            String.format(
+                "%s\n\nSee also #%d and [build log](%s)",
+                rel.body(), issue.number(), home
+            )
+        );
+        issue.comments().post(
+            String.format(
+                CommentsTag.PHRASES.getString("CommentsTag.duplicate"),
+                tag
+            )
+        );
+        Logger.info(this, "duplicate tag %s commented", tag);
+    }
+
+    /**
+     * Valid version numbers:
+     * .1
+     * 2.2
+     * .1.2
+     * 1.2.3.4.5.6.7
+     *
+     * Invalid version numbers:
+     * abc
+     * a.b.c
+     * 1.
+     * 1.2.
+     * @param version Version number from a release
+     * @return True if the version is valid, false otherwise
+     */
+    private static boolean valid(final CharSequence version) {
+        return CommentsTag.VERSION_PATTERN.matcher(version).matches();
+    }
+
+    /**
+     * Is this tagged release valid?  A tagged release is valid if it's greater
+     * than any previous release.
+     * @param tag The release to be tagged
+     * @param previous The previous releases
+     * @return True if the release is valid
+     */
+    private static boolean valid(final String tag,
```

Comment:
@gumbelmj You are overloading the above method in a way that changes the action of the method entirely.
One version checks a string for compliance with a pattern, the other checks whether this string compares correctly to a collection of other strings.